### ISSUE TITLE
[css-values-5] Using `first-valid()` and `mix()` in shorthands

### ISSUE THREAD

**Issue by cdoublev** [w3c/csswg-drafts#11136]
There are currently three `<whole-value>` substitution functions: [`first-valid()`](https://drafts.csswg.org/css-values-5/#first-valid), [`mix()`](https://drafts.csswg.org/css-values-5/#mix), `toggle()`. All are resolved at computed value time or later.

Using them as a shorthand value is specified for `toggle()`, but not for the other two: should they expand like [arbitrary substitutions](https://drafts.csswg.org/css-values-5/#substitution-in-shorthands), with a pending-substitution value?

**Comment by Loirooriol** [comment]
The current definition of `toggle()` on shorthands is broken, see #6764.

This needs to use internal values akin to pending-substitution values, or #8055.

But I think `first-valid()` is resolved at parse time?

**Comment by tabatkins** [comment]
> But I think first-valid() is resolved at parse time?

No, it's not. If you're solely providing literal arguments, then yes, it *could* be, but then you could just write them as separate declarations for the same effect.  The point of first-valid() is to take var()s in their arguments, so in practice it's always resolved at computed value time, during substitution. For consistency, we just make it *always* resolve then, like the other arbitrary-substitution functions.

**Comment by Loirooriol** [comment]
> then you could just write them as separate declarations for the same effect

Wasn't this the behavior that you wanted in https://github.com/w3c/csswg-drafts/issues/5055#issuecomment-944959396 ? There is actually #10989 about this.

**Comment by cdoublev** [comment]
Can you please clarify this, and close #10989?

  > `<first-valid()> = first-valid( <declaration-value># )`
  >
  > If none of the arguments represent a valid value for the property, the property is invalid at computed-value time.

For example:

  > However, if none of them includes a substitution function that resolve after parsing, it is invalid at parse time.

---

So the question in the op is still open for `interpolate()`.
